client/src/App.js

```diff
@@ -10,24 +10,32 @@ import TendersList from "./TenderList";
 import Header from "./Header";
 import Footer from "./Footer";
 import LogInForm from "./pages/LogInForm";
+import { useState } from "react";
 
-const App = () => (
-	<>
-		<Header className="header" />
-		<Routes>
-			<Route path="/" element={<Home />} />
-			<Route path="/publish-tender" element={<PublishTenderForm />} />
-			<Route path="/BuyerTenderList" element={<BuyerTenderList />} />
-			<Route path="/BidderBiddingList" element={<BidderBiddingList />} />
-			<Route path="/signup" element={<SignUp />} />
-			<Route path="/admin-dashboard" element={<AdminDashboard />} />
-			<Route path="/buyer-dashboard" element={<BuyerDashboard />} />
-			<Route path="/list-tenders" element={<TendersList />} />
-			<Route path="/list-tenders/page/:pageNumber" element={<TendersList />} />
-			<Route path="/login" element={<LogInForm />} />
-		</Routes>
-		<Footer className="footer" />
-	</>
-);
+const App = () => {
+	const [role, setRole] = useState(null);
```

Comment:
This is not needed here. We need to read the role from local storage when the header is rendered